### PR TITLE
Added warn-k8s_ocp-deployment_deploymentconfig-bestpractices

### PR DESCRIPTION
**Pull request by garethahealy**
#### What is this PR About?
More splitting: https://github.com/redhat-cop/rego-policies/pull/19

#### How do we test this?
Action output green

cc: @redhat-cop/rego-policies

### CODE REVIEW

File: policy/warn-k8s_ocp-deployment_deploymentconfig-bestpractices.rego [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+  msg := sprintf("%s/%s: replicas is %d - expected replicas to be greater than 1 for HA guarantees.", [input.kind, input.metadata.name, input.spec.replicas])
+}
+
+warn[msg] {
+  apis := ["apps/v1", "apps.openshift.io/v1"]
+  kinds := ["Deployment", "DeploymentConfig"]
+
+  input.apiVersion == apis[_]
+  input.kind == kinds[_]
+  input.spec.replicas % 2 == 0
```
Comment (springdo) [inline policy/warn-k8s_ocp-deployment_deploymentconfig-bestpractices.rego:31]: sweet bit of maths 🤓

File: policy/warn-k8s_ocp-deployment_deploymentconfig-bestpractices.rego [hunk 2]
```diff
@@ -0,0 +1,450 @@
+package main
+
+warn[msg] {
+  apis := ["apps/v1", "apps.openshift.io/v1"]
+  kinds := ["Deployment", "DeploymentConfig"]
+
+  input.apiVersion == apis[_]
+  input.kind == kinds[_]
```
Comment (springdo) [inline policy/warn-k8s_ocp-deployment_deploymentconfig-bestpractices.rego:8]: These above lines are repeated loads, they could probs be abstracted into a function right?

Comment (garethahealy) [reply policy/warn-k8s_ocp-deployment_deploymentconfig-bestpractices.rego:8]: @springdo yep, we can reduce some duplication by creating utils rego files and importing them as 'data'

Comment (springdo) [reply policy/warn-k8s_ocp-deployment_deploymentconfig-bestpractices.rego:8]: I'll raise an issue for that, we can tackle it separately to this monster 🙈

**Pull request merged**